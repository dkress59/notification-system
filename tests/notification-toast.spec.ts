class CustomElement extends window.HTMLElement {
	constructor() {
		super()
		this.attachShadow({ mode: 'open' })
		this.shadowRoot!.innerHTML = '<template><slot /></template>'
	}
}

describe('jest', () => {
	it('is working', () => {
		window.customElements.define('custom-element', CustomElement)

		const defaultElement = document.createElement('custom-element')
		expect(
			defaultElement.shadowRoot!.querySelector('template'),
		).toContainHTML('<slot>')
	})
})
describe('<toast-notification />', () => {
	it('matches snapshot', () => {
		const notification = document.createElement('toast-notification')
		document.body.appendChild(notification)
		expect(document.body.innerHTML).toMatchSnapshot()
		const element = document.body.querySelector('toast-notification')!
		expect(element.shadowRoot?.innerHTML).toMatchSnapshot()
	})
})
