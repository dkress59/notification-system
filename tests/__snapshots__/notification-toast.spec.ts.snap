// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<toast-notification /> matches snapshot 1`] = `"<toast-notification style=\\"transition: none;\\" class=\\"hidden\\"></toast-notification>"`;

exports[`<toast-notification /> matches snapshot 2`] = `
"<style>mock-style {
	mock-prop: value;
}</style>"
`;

exports[`<toast-notification /> matches snapshot 3`] = `
"<style>mock-style {
	mock-prop: value;
}</style>"
`;
